plugins/out_pgsql/pgsql_connections.c

```diff
@@ -73,6 +73,19 @@ void *pgsql_create_connection(struct flb_pgsql_config *ctx)
     flb_plg_info(ctx->ins, "switching postgresql connection "
                  "to non-blocking mode");
 
+    if (ctx->schema != FLB_PGSQL_SCHEMA) {
+        res = PQexec(conn->conn, "SET search_path = %s", ctx->schema);
```

Comment:
The schema here needs to be escaped, otherwise, can produce a some unexpected behavior, including execute not desired SQL queries